runtime/v2/manager.go

```diff
@@ -207,8 +207,20 @@ func (m *TaskManager) Add(ctx context.Context, task runtime.Task) error {
 }
 
 // Delete a runtime task
-func (m *TaskManager) Delete(ctx context.Context, id string) {
+func (m *TaskManager) Delete(ctx context.Context, id string) (*runtime.Exit, error) {
+	task, err := m.tasks.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	shim := task.(*shim)
+	exit, err := shim.delete(ctx, m.tasks.Delete)
+	if err != nil {
+		return nil, err
+	}
+
 	m.tasks.Delete(ctx, id)
```

Comment:
How does this `Delete` call differ from the one called in `shim.delete`?

Reply:
I guess I don't understand the difference between `s.ID()` and the `id` here

Reply:
`ContainerID` = `BundleId` = `TaskId`

https://github.com/containerd/containerd/blob/main/services/tasks/local.go#L210
https://github.com/containerd/containerd/blob/main/runtime/v2/manager.go#L122
https://github.com/containerd/containerd/blob/dd08c7a49fd8a0d5aade30b2a5a864bb61b9ba27/runtime/v2/bundle.go#L59

Reply:
Yeah,  the call to "m.tasks.Delete" at line 222 seems redundant now.